src/NUnitFramework/framework/Internal/TypeWrapper.cs

```diff
@@ -258,5 +259,33 @@ public override string ToString()
         {
             return Type.ToString();
         }
+
+        /// <summary>
+        /// Returns all methods declared by this type that have the specified attribute, optionally
+        /// including base classes. Methods from a base class are always returned before methods from a class that
+        /// inherits from it.
+        /// </summary>
+        /// <param name="inherit">Specifies whether to search the fixture type inheritance chain.</param>
+        public IMethodInfo[] GetMethodsWithAttribute<T>(bool inherit) where T : class
+        {
+            if (!inherit)
+            {
+                return Type
+                                  .GetMethods(Reflect.AllMembers | BindingFlags.DeclaredOnly)
+                                  .Where(method => method.IsDefined(typeof(T), inherit: false))
+                                  .Select(method => new MethodWrapper(Type, method))
+                                  .ToArray();
+            }
+
+            var methodsByDeclaringType = Type
+                                                    .GetMethods(Reflect.AllMembers | BindingFlags.FlattenHierarchy) // FlattenHierarchy is complex to replicate by looping over base types with DeclaredOnly.
+                                                    .Where(method => method.IsDefined(typeof(T), inherit: true))
+                                                    .ToLookup(method => method.DeclaringType);
```

Comment:
Can you clean up the indentation of these to be more like the original in `Reflect.cs`?

Reply:
Done.